aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/Translator.java

```diff
@@ -104,6 +137,12 @@ public static CreateDbInstanceReadReplicaRequest createDbInstanceReadReplicaRequ
                 .tags(Tagging.translateTagsToSdk(tagSet))
                 .useDefaultProcessorFeatures(model.getUseDefaultProcessorFeatures())
                 .vpcSecurityGroupIds(CollectionUtils.isNotEmpty(model.getVPCSecurityGroups()) ? model.getVPCSecurityGroups() : null);
+        if (!ResourceModelHelper.isSqlServer(model)) {
```

Comment:
We need to be very defensive here and make sure we have the coverage for both code paths -> IsSqlServer and IsNotSqlServer on our testing.

Reply:
Same for the other code paths below with a special callout on `if (ResourceModelHelper.isStorageParametersModified(model) && ResourceModelHelper.isSqlServer(model))` which is seems like a separate user story.

Reply:
Both TranslatorTests (see `test_restoreFromAuroraPostgresSnapshotRequest_storageType_shouldBeSetOnUpdate` and `test_restoreFromSqlServerSnapshotRequest_storageType_shouldBeSetOnUpdate`) and CreateHandlerTests verify SQL server and non SQL server paths.

We also have ResourceModelHelperTest which mostly focus on shouldUpdateAfterCreate method.